web/modules/custom/joinup_discussion/src/EventSubscriber/WorkflowTransitionEventSubscriber.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Drupal\joinup_discussion\EventSubscriber;
+
+use Drupal\comment\Plugin\Field\FieldType\CommentItemInterface;
+use Drupal\state_machine\Event\WorkflowTransitionEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Event subscriber to close comments on archived discussions.
```

Comment:
The ticket in stake talks about disabling discussions, not archiving. 
Can we stick with one terminology since the action is also called "Disable"?